src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -94,14 +94,19 @@ internal sealed partial class LocalRewriter : BoundTreeRewriterWithStackGuard
                 // Presence of sequence points in the tree affects final IL, therefore, we always generate them.
                 var localRewriter = new LocalRewriter(compilation, method, methodOrdinal, statement, containingType, factory, previousSubmissionFields, allowOmissionOfConditionalCalls, diagnostics,
                                                       dynamicInstrumenter != null ? new DebugInfoInjector(dynamicInstrumenter) : DebugInfoInjector.Singleton);
-
                 statement.CheckLocalsDefined();
-                var loweredStatement = (BoundStatement)localRewriter.Visit(statement);
+                var visited = localRewriter.Visit(statement);
+                var loweredStatement = (BoundStatement)visited;
                 loweredStatement.CheckLocalsDefined();
                 sawLambdas = localRewriter._sawLambdas;
                 sawLocalFunctions = localRewriter._sawLocalFunctions;
                 sawAwaitInExceptionHandler = localRewriter._sawAwaitInExceptionHandler;
 
+                if (visited is BoundBlock block)
```

Comment:
does it make sense to Debug.Assert about the type or null state of `visited`?

Reply:
Assert fails on a boundstatementlist. Should I allow boundstatementlists to be rewritten as well?

Reply:
Answer per andy: nope